### PR TITLE
Add GET and PUT endpoints for user details

### PR DESCRIPTION
**Pull request by JerryAgbesi**

### CODE REVIEW

**Review by benessilfie — DISMISSED** [review]
Test

File: api/api_models/user_response.py [hunk 1]
```diff
@@ -7,7 +8,22 @@ class UserResponse(BaseModel):
...
         orm_mode = True
+
+class ProfileResponse(BaseModel):
+    email: Optional[EmailStr] 
+    first_name: Optional[str] 
+    last_name: Optional[str] 
+    github_profile: Optional[str] 
+    twitter_profile: Optional[str] 
+    linkedin_profile: Optional[str] 
+    portfolio_url: Optional[str]
```
Comment (blitzblade) [inline api/api_models/user_response.py:28]: Clean up model name and how it's used.
Also consider refactoring existing model that could be reused here.

File: api/routes/profile_page.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            user_exists.email = userDetails.email if userDetails.email else user_exists.email
+            user_exists.first_name = userDetails.first_name if userDetails.first_name else user_exists.first_name
+            user_exists.last_name = userDetails.last_name if userDetails.last_name else user_exists.last_name
+            user_exists.github_profile = userDetails.github_profile
+            user_exists.twitter_profile = userDetails.twitter_profile
+            user_exists.linkedin_profile = userDetails.linkedin_profile
+            user_exists.portfolio_url = userDetails.portfolio_url
+            user_exists.profile_pic_url = userDetails.profile_pic_url
+
+            db.commit()
```
Comment (blitzblade) [inline api/routes/profile_page.py:32]: Clean this up together with the model changes we talked about.

File: api/api_models/user_sign_up.py [hunk 3]
```diff
@@ -5,6 +5,11 @@ class UserSignUp(BaseModel):
     email: EmailStr = Field(...)
     first_name: str = Field(..., min_length=2)
     last_name: str = Field(..., min_length=2)
+    github_profile: str=Field('')
+    twitter_profile: str=Field('')
+    linkedin_profile: str=Field('')
+    portfolio_url: Optional[str]=Field('')
+    profile_pic_url: str=Field('')
```
Comment (blitzblade) [inline api/api_models/user_sign_up.py:12]: These should be nullable instead of having default empty string

**Review by benessilfie — APPROVED** [review]

**Review by eeTiboah — APPROVED** [review]

File: test/test_profile_page.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from fastapi.testclient import TestClient
+from api.api_models.user import UserResponse
+from app import app
+
+testclient = TestClient(app)
+
+def test_get_profile():
+    res = testclient.get("/api/v1/users/profile/?id=1")
+    
+    assert res.status_code == 200
```
Comment (blitzblade) [inline test/test_profile_page.py:10]: Check fields to be sure it matches what you expect

**Review by blitzblade — APPROVED** [review]

**Pull request merged**